apispec/openapi-model/src/main/scala/sttp/tapir/openapi/OpenAPI.scala

```diff
@@ -169,60 +309,122 @@ object ParameterStyle extends Enumeration {
+  def extensions(updated: ListMap[String, ExtensionValue]): RequestBody = copy(extensions = updated)
+  def required(boolean: Boolean): RequestBody = copy(required = Some(boolean))
+  def addExtension(key: String, value: ExtensionValue): RequestBody = copy(extensions = extensions.updated(key, value))
+}
+object RequestBody {
+  val Empty: RequestBody = RequestBody()
+}
 
-case class MediaType(
+final case class MediaType(
     schema: Option[ReferenceOr[Schema]] = None,
     example: Option[ExampleValue] = None,
     examples: ListMap[String, ReferenceOr[Example]] = ListMap.empty,
     encoding: ListMap[String, Encoding] = ListMap.empty,
     extensions: ListMap[String, ExtensionValue] = ListMap.empty
-)
+) {
+  def schema(updated: Schema): MediaType = copy(schema = Some(Right(updated)))
+  def example(updated: ExampleValue): MediaType = copy(example = Some(updated))
+  def examples(updated: ListMap[String, ReferenceOr[Example]]): MediaType = copy(examples = updated)
+  def addExample(key: String, updated: Example): MediaType = copy(examples = examples.updated(key, Right(updated)))
+  def extensions(updated: ListMap[String, ExtensionValue]): MediaType = copy(extensions = updated)
+  def addExtension(key: String, value: ExtensionValue): MediaType = copy(extensions = extensions.updated(key, value))
+}
 
-case class Encoding(
+object MediaType {
+  val Empty: MediaType = MediaType()
+}
+
+final case class Encoding(
     contentType: Option[String] = None,
     headers: ListMap[String, ReferenceOr[Header]] = ListMap.empty,
     style: Option[ParameterStyle.ParameterStyle] = None,
     explode: Option[Boolean] = None,
     allowReserved: Option[Boolean] = None,
     extensions: ListMap[String, ExtensionValue] = ListMap.empty
-)
+) {
+  def contentType(updated: String): Encoding = copy(contentType = Some(updated))
+  def headers(updated: ListMap[String, ReferenceOr[Header]]): Encoding = copy(headers = updated)
+  def addHeader(key: String, header: Header): Encoding = copy(headers = headers.updated(key, Right(header)))
+  def style(updated: ParameterStyle.ParameterStyle): Encoding = copy(style = Some(updated))
+  def explode(updated: Boolean): Encoding = copy(explode = Some(updated))
+  def allowReserved(updated: Boolean): Encoding = copy(allowReserved = Some(updated))
+  def extensions(updated: ListMap[String, ExtensionValue]): Encoding = copy(extensions = updated)
+  def addExtension(key: String, value: ExtensionValue): Encoding = copy(extensions = extensions.updated(key, value))
+}
+
+object Encoding {
+  val Empty: Encoding = Encoding()
+}
 
 sealed trait ResponsesKey
 case object ResponsesDefaultKey extends ResponsesKey
-case class ResponsesCodeKey(code: Int) extends ResponsesKey
+final case class ResponsesCodeKey(code: Int) extends ResponsesKey
 
 // todo: links
-case class Response(
-    description: String,
+final case class Response(
+    description: Option[String] = None,
```

Comment:
`description` is required on a response: https://swagger.io/specification/#responses-object

Reply:
you are right. will fix